### ISSUE TITLE
Nvidia Geforce GT 525 M

### ISSUE THREAD

**Issue by CritSin** [fireice-uk/xmr-stak#2024]
support r not?

**Comment by psychocrypt** [comment]
you must compile it by hand if this is a fermi gpu. else update your driver
to 9.X

**Comment by CritSin** [comment]
My nvidia deiver upgraded but not work again

**Comment by Spudz76** [comment]
I have NVS 5400M which is nearly identical, arch 2.1, and you must compile with CUDA 8.0 SDK, set `CUDA_ARCH=20` for best results.

You can't run newer than last CUDA 8.0 driver, which was `386.28` for Windows (you never told us your OS) 390 and 400 series will never work (other than for graphics)
Newer drivers than that only run the 2D/3D with no CUDA, unless you built with cudart 8.0 (cudart64_80.dll) support, which is the only way to run old CUDA on newer drivers (but I have less luck backward compatible, and you have to add custom link flags for cudart, just run the exactly matching driver)

**Comment by Spudz76** [comment]
Also pending accept, new docs section [here](https://github.com/Spudz76/xmr-stak/blob/dev-cudaDocs/doc/compile.md#cuda-runtime-versus-cuda-sdk) summarizing the above

**Comment by CritSin** [comment]
Please write link for downloading best cuda nvidia gt 525m

**Comment by Spudz76** [comment]
Windows?

Linux?

**Comment by Spudz76** [comment]
[Win7 Driver](https://www.nvidia.com/download/driverResults.aspx/123940/en-us)
[Win10 Driver](https://www.nvidia.com/download/driverResults.aspx/123938/en-us)

[SDK 8.0](https://developer.nvidia.com/cuda-80-ga2-download-archive) - assuming you can navigate from there based on your OS

**Comment by Zenitur** [comment]
I have Linux and NVIDIA GeForce 560 Ti. If my `xmrstak_cuda_backend.so` was compiled with CUDA 9.1, miner cannot start to mine.
```
[2018-11-01 03:29:25] : Mining coin: cryptonight_v8
[2018-11-01 03:29:25] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda10_0'
WARNING: NVIDIA Insufficient driver!
WARNING: NVIDIA no device found
[2018-11-01 03:29:25] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda9_2'
WARNING: NVIDIA Insufficient driver!
WARNING: NVIDIA no device found
[2018-11-01 03:29:25] : NVIDIA: try to load library 'xmrstak_cuda_backend'
NVIDIA: found 1 potential device's
[2018-11-01 03:29:26] : Starting NVIDIA GPU thread 0, no affinity.
CUDA [9.1/9.1] GPU#0, device architecture 21: "GeForce GTX 560 Ti"... WARNING: skip device - binary does not contain required device architecture
[2018-11-01 03:29:26] : Setup failed for GPU 0. Exiting.
```
But if it was compiled with CUDA 8.0, it's starting to mine:
```
[2018-11-01 03:31:29] : Mining coin: cryptonight_v8
[2018-11-01 03:31:29] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda10_0'
WARNING: NVIDIA Insufficient driver!
WARNING: NVIDIA no device found
[2018-11-01 03:31:29] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda9_2'
WARNING: NVIDIA Insufficient driver!
WARNING: NVIDIA no device found
[2018-11-01 03:31:29] : NVIDIA: try to load library 'xmrstak_cuda_backend'
NVIDIA: found 1 potential device's
[2018-11-01 03:31:29] : Starting NVIDIA GPU thread 0, no affinity.
CUDA [9.1/8.0] GPU#0, device architecture 21: "GeForce GTX 560 Ti"... device init succeeded
[2018-11-01 03:31:29] : NVIDIA: use library 'xmrstak_cuda_backend'
[2018-11-01 03:31:29] : MEMORY ALLOC FAILED: mlock failed
[2018-11-01 03:31:30] : WARNING: No AMD OpenCL platform found. Possible driver issues or wrong vendor driver.
[2018-11-01 03:31:30] : WARNING: backend AMD (OpenCL) disabled.
[2018-11-01 03:31:30] : Fast-connecting to stratum+tcp://xmr.pool.minergate.com:45700 pool ...
[2018-11-01 03:31:30] : Pool stratum+tcp://xmr.pool.minergate.com:45700 connected. Logging in...
[2018-11-01 03:31:30] : Difficulty changed. Now: 1063.
[2018-11-01 03:31:30] : Pool logged in.
[2018-11-01 03:31:33] : New block detected.
```
My build: https://yadi.sk/d/i7IR3eUxvsjx8A (don't afraid of this link, this is a cloud service)

**Comment by Spudz76** [comment]
Yeah, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0

Best results run the last CUDA 8.0 driver 386.28 as I have also said (yours as you see is 9.1 running in 8.0 compatibility mode... compatibility mode is never optimal in anything)

**Comment by vtec888** [comment]
hi Spudz76m, i am newbie to mining..recently i have stuck with the setup also almost same issue with above..
if you dont mind can you guide us from driver install, what item to download such as cuda version, which visual studio version required and etc  in-order to let the miner start work.. 
My current laptop in window 10,  geforce gtx 540m  before i format my laptop that was fine..these have stuck me for quite sometimes...

Thanks

Des

**Comment by Spudz76** [comment]
All of that is standard, in the docs, other than the table of driver-to-cuda is not included yet
For CUDA8 you must install v140 Toolset in addition to normal MSVSCE-2017 (but it says so in the docs)

**Comment by mechanator** [comment]
> Yeah, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0, Fermi requires CUDA 8.0
> 
> Best results run the last CUDA 8.0 driver 386.28 as I have also said (yours as you see is 9.1 running in 8.0 compatibility mode... compatibility mode is never optimal in anything)

Then who can make a windows binary latest release with CUDA 8 for older Fermi cards? Assuming you have the correct drivers for Win 10 for Telsa M2050 through M2090 and GTX 400/500 series. It would be nice to mine CN v2 and newer variants.   XMRIG did it. 
One thing I noticed is that the Tesla cards have 3-6GB of ram, but are thread limited. Even the  GTX 465 has 1.5GB, and the GTX 580 / 590 has 1.5GB X 2 with two cores.   But the amount of ram used by the cuda call is much smaller than that.

**Comment by Spudz76** [comment]
> Then who can make a windows binary latest release with CUDA 8 for older Fermi cards?
Literally anyone.  I have never run a Release binary there is too much incentive to self-compile, and it is pretty easy to rebuild new pulls once the toolchain/SDK stack is set up.

The way it builds releases now could be extended to make a Release binary pack for all CUDA major versions (8, 9.2, 10.0), which I think was a goal related to the recent CUDA-backend-dll multiloader but the CMake/AppVeyor parts need to be completed.

**Comment by psychocrypt** [comment]
No since cuda 8 requires v140 and all other v141 I can not mix it :-(

**Comment by mechanator** [comment]
Before the last reply, I took an attempt to compile it with the instructions supplied. It comes up with this error message which I tracked back to the jconf.cpp CPU threads being messed up saying "Thread 0 has invalid config."
See screenshot here: https://imgur.com/GD4ioDE

  It looks similar to the same error message string in the amd section of jconf.cpp.
So I made a dare and modified the message and to make it to return true instead of false at:
"for(size_t i=0; i < GetThreadCount(); i++)
	{
		if(!GetThreadConfig(i, c))
		{
			printer::inst()->print_msg(L0, "Thread %llu has invalid CPU config.", int_port(i));
			return true;
		}
	}

	return true;
}
"

With that change. I still get cpu thread errors,
See attached screenshot: https://imgur.com/jD3Q3nS
Some kind of warning about affinity settings.
Although, it will mine using GPUs only with the --noCPU option or commenting out the options in cpu.txt.  
see screenshot: https://imgur.com/DJcCstD

System Build information: 
i5 2500k Asus p8z77 motherboard, Cpu is AES-Ni, BIOS last rev 1402, 
Windows 10 new build a week old.
Pulled the git code for 2.7.1 on 12/15
Installed either driver 385.28 for gtx 400/500 and Tesla support, installed CUDA 8.1.6.2 , 
Same error message with CUDA8 custom driver but that doesn't run the GTX cards.
Same error message on 386.28 for Tesla only cards.
Followed your compilation instructions exactly.
Installed VS 2017 Community with the options mentioned.
Tested version: [cmake 3.9](https://cmake.org/files/v3.9/cmake-3.9.0-rc3-win64-x64.msi) installed.
Installed that AMD OCL Light kit,. from here: https://github.com/GPUOpen-LibrariesAndSDKs/OCL-SDK/releases
I did install this:
### Dependencies OpenSSL/Hwloc and Microhttpd
- For CUDA 8*:
  - Download version 1 of the precompiled binary from https://github.com/fireice-uk/xmr-stak-dep/releases/download/v1/xmr-stak-dep.zip
  - Version 1 of the pre-compiled dependencies is not compatible with Visual Studio Toolset v141
- Extract archive to `C:\xmr-stak-dep`  - ok did that.
The build script I made:
---------------------------
mkdir build
cd build
set CMAKE_PREFIX_PATH=C:\xmr-stak-dep\hwloc;C:\xmr-stak-dep\libmicrohttpd;C:\xmr-stak-dep\openssl
cmake -G "Visual Studio 15 2017 Win64" -T v140,host=x64 ..
cmake --build . --config Release --target install
cd bin\Release
copy C:\xmr-stak-dep\openssl\bin\* .
--------------------------------------

I also tried excluding components with the cmake switches to no avail still error messages about affinity settings out range  when mining on CPU combined.
Using any of   -DMICROHTTPD_ENABLE=OFF -DOpenSSL_ENABLE=OFF  -DOpenCL_ENABLE=ON -DCMAKE_LINK_STATIC=ON    , Didn't make a difference.

These card I know work since I can run xmr-stak 2.45 with cuda 9.2 , but want the variant 2 algos.

At least the opencl mode runs, but at performance hit, but CUDA8 for arqma v7 cn-lite was only 10% slower.
I hope this helps figuring out what causes the cpu thread crash.

**Comment by Spudz76** [comment]
Did you delete all config files and regenerate?  Many new (and required) options were added and will not parse "old version" configs.  Check topmost line of each config file, it will say what version had created it.  If it doesn't say any version then it's REEEEAL old config file format, definitely won't work.

**Comment by mechanator** [comment]
You are correct sir. Restarted from scratch. It runs without the strange thread error.
The good news is that you can still insall CUDA 8 Toolkit, the older drivers, and still run XMR-STAK 2.7.1 with  all the latest updates.

**Comment by Spudz76** [comment]
The config file parsing could be a bit more helpful when it can't find a required setting in your current JSON structure.  Partially the reason for all-or-nothing behavior is that RapidJSON is called with the whole content of the file and returns the result (or fails).  There is not much per-line state happening to be able to tell what's wrong.

**Comment by mechanator** [comment]
I buillt the CUDA 8 for 2.71 and made a release for it here:
https://github.com/mechanator/XMR-STAK-CUDA-8-2.7.x
